src/libs/ReportUtils.ts

```diff
@@ -11522,73 +11526,6 @@ function prepareOnboardingOnyxData(
         guidedSetupData.push({type: 'message', ...textMessage});
     }
 
-    let selfDMParameters: SelfDMParameters = {};
-    if (engagementChoice === CONST.ONBOARDING_CHOICES.PERSONAL_SPEND || engagementChoice === CONST.ONBOARDING_CHOICES.TRACK_WORKSPACE) {
-        const selfDMReportID = findSelfDMReportID();
-        let selfDMReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${selfDMReportID}`];
-        let createdAction: ReportAction;
-        if (!selfDMReport) {
-            const currentTime = DateUtils.getDBTime();
-            selfDMReport = buildOptimisticSelfDMReport(currentTime);
```

Comment:
At this time users will have selfDM created although it won't be in their onyx. Probably instead of creating it optimistically it would be better to return it in `CompleteGuidedSetup`